### ISSUE TITLE
TypeError: server must be web.Application TestServer instance, found type: <class 'aiohttp.web.Application'>

### ISSUE THREAD

**Issue by argaen** [aio-libs/aiohttp#2929]
## Long story short

Can't create a `TestClient` instance with a `web.Application` instance. According to the docs (https://docs.aiohttp.org/en/stable/testing.html#pytest) and exception message it should be possible but in code aiohttp just checks if its an instance of `BaseTestServer`: https://github.com/aio-libs/aiohttp/blob/master/aiohttp/test_utils.py#L161

So either the docs or the implementation are wrong. Tried with `aiohttp==3.1.2` (also reproduced with older versions).

## Steps to reproduce

```
Python 3.6.0+ (3.6:1555e7776321, Jun 27 2017, 18:36:56)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from aiohttp import web
ap>>> app = web.Application()
>>> from aiohttp.test_utils import TestClient
>>> TestClient(app)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/manuelmiranda/.pyenv/versions/bb36/lib/python3.6/site-packages/aiohttp/test_utils.py", line 193, in __init__
    "instance, found type: %r" % type(server))
TypeError: server must be web.Application TestServer instance, found type: <class 'aiohttp.web.Application'>
>>>
```

**Comment by argaen** [comment]
I guess the usage has to be `TestClient(TestServer(app))`?

**Comment by asvetlov** [comment]
Thanks for the report.
Documentation should be fixed, `web.Application` should be dropped from the error message.

**Comment by asvetlov** [comment]
Yes, `TestClient(TestServer(app))`

**Comment by argaen** [comment]
You can assign the issue to me. Will provide MR to update docs and exception message today or tomorrow

**Comment by asvetlov** [comment]
Thank you

**Comment by argaen** [comment]
@asvetlov I was double checking `TestClient` docs (https://docs.aiohttp.org/en/stable/testing.html#test-client). They say the following:
```
app_or_server –
BaseTestServer instance for making client requests to it.

If the parameter is aiohttp.web.Application the tool creates TestServer implicitly for serving the application.
```
I think its worth in the constructor of `TestClient` adding the code that if its a `web.Application` object, to convert it to a TestServer. Same as the pytest fixture is doing right now: https://github.com/aio-libs/aiohttp/blob/master/aiohttp/pytest_plugin.py#L304

In fact if we move this logic into `TestClient` we can remove the logic from the fixture. What do you think?

**Comment by asvetlov** [comment]
I think we do the conversion in both pytest fixture and unittest base test case.
Adding the check into `TestClient` overcomplicates the code, that's why the support was removed.

**Comment by argaen** [comment]
> I think we do the conversion in both pytest fixture and unittest base test case.

Yeah, its in https://github.com/aio-libs/aiohttp/blob/master/aiohttp/test_utils.py#L350.

Wouldn't it be better to have it centralized in the `TestClient` constructor? This way you ensure no matter what you use, the behavior is the same. If not, you need this "duplication" in pytest fixtures, unittest (or any other testing framework people may use).

**Comment by asvetlov** [comment]
I'm ok with duplication.
Otherwise `TestClient.__init__` becames too cumbersome.
I've tried this way and rejected it.
In simple cases a user should never create `TestClient` explicitly but use pytest/unittest tools.
In complex scenarios I see nothing wrong with the explicit creation of proper `TestServer` instance first.

**Comment by argaen** [comment]
Okay, fine with it :)

> In simple cases a user should never create TestClient explicitly but use pytest/unittest tools.

Our case is simple though, we just want to create the app once per test so we had to rewrite the pytest fixture giving it a `session` scope

**Comment by asvetlov** [comment]
By using `session` scoped fixtures you are breaking the test isolation principle :)

**Comment by argaen** [comment]
Agreed, but we have lots of acceptance test and our app start up takes lots of time because of cache warming. There is probably a better way but for now it works (with its risks.. 😓)

**Comment by asvetlov** [comment]
Fixed by #2933
